receiver/googlecloudmonitoringreceiver/internal/metrics_conversion.go

```diff
@@ -226,10 +226,34 @@ func (mb *MetricsBuilder) ConvertDistributionToMetrics(ts *monitoringpb.TimeSeri
 			continue
 		}
 
-		countTotal := uint64(0)
 		targetBucketCounts := targetDataPoint.BucketCounts()
-		targetBucketCounts.EnsureCapacity(len(sourceBucketCounts))
-		for _, bucketCount := range sourceBucketCounts {
+
+		// The number of elements in bucket_counts array must be by one greater than the number of elements in explicit_bounds array.
+		lenBounds := targetDataPoint.ExplicitBounds().Len()
+		lenCounts := lenBounds + 1
+
+		// The exception to this rule is when the length of bucket_counts is 0, then the length of explicit_bounds must also be 0.
+		// Google says this case should never happen, but you never know...
+		if lenBounds == 0 {
+			// A Histogram without buckets conveys a population in terms of only the sum and count, and may be interpreted as a
+			// histogram with single bucket covering (-Inf, +Inf).
+			targetDataPoint.SetCount(uint64(distributionValue.GetCount()))
```

Comment:
We should always get the count from the distribution's count, and stop trying to sum the total bucket counts below.

Reply:
Ah, that makes sense.  Leaving it as-is is fine.